src/Database/Bolt/Transaction.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Database.Bolt.Transaction
+  ( TxError
+  , transact
+  , transact_
+  )
+where
+
+import           Database.Bolt.Connection.Pipe
+import           Database.Bolt.Connection.Type
+import           Database.Bolt.Connection       ( BoltActionT
+                                                , run
+                                                )
+import           Database.Bolt.Record
+
+import           Control.Exception              ( IOException )
+import           Control.Monad                  ( join )
+import           Control.Monad.Catch            ( Exception
+                                                , catch
+                                                , throwM
+                                                )
+import           Control.Monad.IO.Class         ( MonadIO(..)
+                                                , liftIO
+                                                )
+import           Control.Monad.Trans.Reader     ( ask )
+import           Data.Functor                   ( void )
+import           Data.Map.Strict                ( empty )
+import           Data.Text                      ( Text )
+
+data TxError = TxError deriving Show
+instance Exception TxError
+
+-- |Runs a list of 'BoltActionT' as an atomic operation and returns all the records or throws 'TxError' if something went wrong
+transact :: [BoltActionT IO [Record]] -> BoltActionT IO [Record]
```

Comment:
If we want this to be `MonadIO m` instead if `IO` I guess that we need to include an [extra package](https://hackage.haskell.org/package/MonadCatchIO-transformers-0.3.1.3/docs/Control-Monad-CatchIO.html). Not sure whether it's worth the trouble.